### ISSUE TITLE
Unable to load dynamic library '/usr/lib/php5/20100525/amqp.so' on Ubuntu

### ISSUE THREAD

**Issue by tlshaheen** [pdezwart/php-amqp#151]
I'm having a similar issue to https://github.com/pdezwart/php-amqp/issues/87. However, the steps he used to fix it have not fixed the issue for me.

After running all the installs, I get a `PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/lib/php5/20100525/amqp.so' - /usr/lib/php5/20100525/amqp.so: undefined symbol: amqp_empty_table in Unknown on line 0` error when I try to load PHP.

What am I missing/doing wrong?

My environment: 
Ubuntu 12.04.5 LTS
PHP 5.4.37
`PHP Api Version:         20100412`
`Zend Module Api No:      20100525`
`Zend Extension Api No:   220100525`
`cmake version 2.8.7`

I've installed lib-rabbitmq following the steps [on a walk-through](http://www.thegeekstuff.com/2013/05/amqp-php-extension/). I used the tarball for [rabbitmq-c v0.5.2](https://github.com/alanxz/rabbitmq-c/releases/tag/v0.5.2). I then went and got [amqp v1.6.0beta2](http://pecl.php.net/package/amqp/1.6.0beta2).

Ran `sudo tar -xvzf amqp-1.6.0beta2`, ran `sudo phpize`, and then ran `sudo ./configure --with-amqp`, `sudo make`, and `sudo make install`. The outputs from those are here: https://gist.github.com/tlshaheen/c283d123e75b5684fd44.

My `make install` never prompts me to add the `amqp.so` extension; is that a sign something went wrong?

My `ls /usr/lib` shows `librabbitmq.so.0` and `librabbitmq.so.0.0.0`. 
`ls /usr/local/lib` shows `librabbitmq.so`. 
`ls /usr/local/lib/lib/x86_64-linux-gnu` shows `librabbitmq.so  librabbitmq.so.1  librabbitmq.so.1.2.2`.

Please let me know what other logs/outputs you need from me.

**Comment by pinepain** [comment]
Too much of `librabbitmq.so`. Looks like one from `/usr/local/lib` linked (`-L/usr/local/lib -lrabbitmq`). I would recommend you to remove all librabbitmq libs and headers and reinstall it from scratches.

**Comment by tlshaheen** [comment]
@pinepain I'll do that and get you fresh install logs. I'll remove all the librabbitmq files from `/usr/lib`, `/usr/local/lib`, and `/usr/local/lib/lib/x86_64-linux-gnu`. Where else should I remove?

**Comment by tlshaheen** [comment]
@pinepain Ok, I cleared out all the librabbit files that `find -iname "*rabbit*"` could find in `/usr`.

Re-installed rabbitmq-c v0.5.2 and amqp v.1.6.0beta2.

I'm now back to the original error I had that lead me to #87 : `PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/lib/php5/20100525/amqp.so' - librabbitmq.so.1: cannot open shared object file: No such file or directory in Unknown on line 0`

You can see my new install logs here: https://gist.github.com/tlshaheen/4290fe38e2c4a3a86f18

**Comment by pinepain** [comment]
I really can't get what is exactly wrong, but looks like the problem is that `librabbitmq` is not properly installed, at least `php-amqp` tries to load it from `/usr/local/lib`, I guess. As you have extension compiled, it means that header files are successfully located. You can try to link  `librabbitmq.so` and company from `/usr/local/lib/x86_64-linux-gnu` to `/usr/local/lib` and if it work, then try to figure out why they are not installed in a proper location (from your gist I assume you tried to install libs to `/usr/local/lib`). You can try to install rabbitmq as we do it in tests, with autotools (look into [`.travis.yml`](https://github.com/pdezwart/php-amqp/blob/master/.travis.yml#L35) file).

**Comment by tlshaheen** [comment]
@pinepain The symbolic link did it. I made the stupid mistake at first with https://github.com/pdezwart/php-amqp/issues/87#issuecomment-36344687 and just copy and pasted his symbolic link code; not realizing my filepath was different, and that I needed to make a change.

Running `sudo ln -s /usr/local/lib/x86_64-linux-gnu/librabbitmq.so.1 /usr/local/lib/librabbitmq.so.1` and restarting Apache fixed the issue for me.

However, something must be off in the install files somewhere for librabbit for my type of environment, since I'm the second one to have this file path issue.

Thanks!
